cmd/ipfs-gateway-fs/main.go

```diff
@@ -53,13 +65,19 @@ func run() error {
 			return err
 		}
 	}
-
 	repo := fsrepo.At(repoPath)
 	if err := repo.Open(); err != nil { // owned by node
 		return err
 	}
-
-	node, err := core.NewIPFSNode(ctx, core.Online(repo))
+	s3, err := makeS3Datastore()
+	if err != nil {
+		return err
+	}
+	enhanced, err := enhanceDatastore(s3)
+	if err != nil {
+		return err
+	}
+	node, err := core.NewIPFSNode(ctx, core.OnlineWithRouting(repo, corerouting.GrandCentralServer(enhanced)))
```

Comment:
Ah, okay. may be worth adding a note here too. given this is the `ipfs-gateway-fs` which we may direct other users to setup.

It may be worth actually splitting this into a different binary, as you had it earlier.